03_letters/n0npax/main_test.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLetters(t *testing.T) {
+	testCases := map[string]struct {
+		input    string
+		expected map[rune]int
+	}{
+		"polish": {input: "dzień dobry",
+			expected: map[rune]int{'y': 1, ' ': 1, 'd': 2, 'z': 1, 'i': 1, 'e': 1, 'ń': 1, 'o': 1, 'b': 1, 'r': 1}},
+		"russian": {input: "доброе утро",
+			expected: map[rune]int{'о': 3, 'р': 2, 'т': 1, 'у': 1, ' ': 1, 'е': 1, 'б': 1, 'д': 1}},
+		"english": {input: "good morning",
+			expected: map[rune]int{'o': 3, 'g': 2, 'd': 1, 'm': 1, 'r': 1, 'n': 2, 'i': 1, ' ': 1}},
+		"chineese": {input: "早上好",
+			expected: map[rune]int{'早': 1, '上': 1, '好': 1}},
+	}
+
+	for name, test := range testCases {
+		t.Run(name, helperTestLetter(test.input, test.expected))
+	}
+}
+
+func helperTestLetter(input string, expected map[rune]int) func(*testing.T) {
+	return func(t *testing.T) {
+		result := letters(input)
+		for k, count := range result {
```

Comment:
I just realised there is another case which you should guard against in this test, you should check the length of the result against the length of the expected. 
If there were a case where `result := { 'a': 1, 'b': 2 }` and `expected := { 'a': 1, 'b': 2, 'c': 3 }` the below wouldn't be able to catch it.

```go
result := letters(input)
if len(result) != len(expected) {
  t.Errorf("some error msg here ... ")
}
for k, count := range result {
  ...
}

Reply:
Fixed. Thanks